data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleFactoryImpl.java

```diff
@@ -336,65 +332,10 @@ protected KafkaConsumer<Object, CloudEvent> createConsumer(final Vertx vertx,
     );
   }
 
-  private CloudEventSender createConsumerRecordSender(
-    final Vertx vertx,
-    final String target,
-    final EgressConfig egress) {
-
-    final var circuitBreakerOptions = createCircuitBreakerOptions(egress);
-    final var circuitBreaker = CircuitBreaker
-      .create(target, vertx, circuitBreakerOptions)
-      .retryPolicy(computeRetryPolicy(egress))
-      .openHandler(r -> logger.info("Circuit breaker opened {}", keyValue("target", target)))
-      .halfOpenHandler(r -> logger.info("Circuit breaker half-opened {}", keyValue("target", target)))
-      .closeHandler(r -> logger.info("Circuit breaker closed {}", keyValue("target", target)));
-
-    return new WebClientCloudEventSender(
-      WebClient.create(vertx, this.webClientOptions), circuitBreaker, circuitBreakerOptions, target
-    );
-  }
-
-  private static CircuitBreakerOptions createCircuitBreakerOptions(final DataPlaneContract.EgressConfig egressConfig) {
-    CircuitBreakerOptions options = new CircuitBreakerOptions();
-
-    // This disables circuit breaker notifications on the event bus
-    options.setNotificationAddress(null);
-
-    if (egressConfig != null) {
-      // Single request timeout
-      options.setTimeout(
-        egressConfig.getTimeout() > 0 ?
-          egressConfig.getTimeout() :
-          DEFAULT_TIMEOUT_MS
-      );
-
-      // Retry options
-      if (egressConfig.getRetry() > 0) {
-        options
-          // TODO reset timeout should be configurable or, at least, set by the control plane
-          .setResetTimeout(
-            egressConfig.getBackoffDelay() > 0 ?
-              egressConfig.getBackoffDelay() :
-              CircuitBreakerOptions.DEFAULT_RESET_TIMEOUT
-          )
-          // TODO max failures should be configurable or, at least, set by the control plane
-          .setMaxFailures(egressConfig.getRetry() * 2)
-          .setMaxRetries(egressConfig.getRetry());
-      }
-    }
-    return options;
-  }
-
-  /* package visibility for test */
-  static Function<Integer, Long> computeRetryPolicy(final EgressConfig egress) {
```

Comment:
I have sympathies for moving this out of the factory, to the actual impl of the sender